Cargo.toml

```diff
@@ -15,28 +15,51 @@ edition = "2018"
 all-features = true
 
 [dependencies]
+# These two are generally really common simple dependencies so it does not seem
+# much of a point to optimize these, but these could in theory be removed for
+# an indirection through std::Formatter.
 dtoa = "0.4"
 itoa = "0.4.3"
+
+# This is a dependency that already exists in url
 percent-encoding = "2.1"
-sha1 = ">= 0.2, < 0.7"
+
+# We need this for redis url parsing
 url = "2.1"
-combine = "3.8.1"
+
+# We need this for script support
+sha1 = { version = ">= 0.2, < 0.7", optional = true }
+
+# Dependency that is shared with tokio.  If we manage to make tokio optional we
+# could also kll this.
 bytes = "0.5"
+
+# These are needed for parsing at the moment 
+combine = "3.8.1"
 futures-util = { version = "0.3.0", features = ["sink"], default-features = false }
-futures-executor = "0.3.0"
+futures-executor = { version = "0.3.0", default-features = false }
+
+# Only needed for AIO
 pin-project-lite = { version = "0.1", optional = true }
-tokio-util = { version = "0.2", features = ["codec"] }
-tokio = "0.2"
+tokio-util = { version = "0.2", features = ["codec"], optional = true }
+
+# Only needed for the r2d2 feature
+r2d2 = { version = "0.8.8", optional = true }
+
+# Only needed for cluster
 crc16 = { version = "0.4.0", optional = true }
 rand = { version = "0.7.0", optional = true }
-r2d2 = { version = "0.8.8", optional = true }
+
+# Unfortunately currently a necessary dependency
```

Comment:
What's the reason we can't put this behind a feature just yet?

Reply:
We're using some traits for it that the parser needs.